### PR TITLE
Support parallel local builds (defaults to sequential)

### LINKED ISSUE #1380: Add parallel local image building.
#248 added parallel building to gcb+kaniko, but not local builds.

### Expected behavior
That local builds use InParallel

### Actual behavior
Local builds use InSequence

### PR DESCRIPTION
**Pull request by dgageot**
Had to cherry pick the schema change from #3453

Fixes #1380

### CODE REVIEW

File: pkg/skaffold/schema/defaults/defaults.go [hunk 1]
```diff
@@ -103,7 +107,21 @@ func defaultToKubectlDeploy(c *latest.SkaffoldConfig) {
 	c.Deploy.DeployType.KubectlDeploy = &latest.KubectlDeploy{}
 }
 
-func withCloudBuildConfig(c *latest.SkaffoldConfig, operations ...func(kaniko *latest.GoogleCloudBuild)) {
+func withLocalBuild(c *latest.SkaffoldConfig, operations ...func(*latest.LocalBuild)) {
+	if gcb := c.Build.LocalBuild; gcb != nil {
```
Comment (briandealwis) [inline pkg/skaffold/schema/defaults/defaults.go]: gcb?

**Comment by loosebazooka** [comment]
Is it possible to selectively disable parallel builds?

File: pkg/skaffold/schema/defaults/defaults.go (hunk 1 above)
Comment (dgageot) [reply pkg/skaffold/schema/defaults/defaults.go]: Thanks! It's fixed now

**Comment by dgageot** [comment]
@loosebazooka selectively how? With this PR, local parallel builds are disabled by default. Users can activate them by changing the skaffold.yaml.

**Comment by loosebazooka** [comment]
I see, can each builder decide if it wants to participate in parallel builds?

**Comment by dgageot** [comment]
@loosebazooka not for now

**Comment by dgageot** [comment]
@nkubala This is documented on that page. See http://35.236.51.86:1313/docs/pipeline-stages/builders/

**Comment by nkubala** [comment]
@dgageot well, the `concurrency` flag is documented, but there isn't really anything saying "we support building locally in parallel, but it's disabled by default". it would be nice to say something like that so people know the feature exists

**Comment by dgageot** [comment]
Let’s do that. Can you merge this one and I’ll edit the doc tomorrow for both local and remote builders?

**Comment by nkubala** [comment]
:+1:

**Pull request merged**